src/api/goods.ts

```diff
@@ -3,17 +3,34 @@ import { Good } from '../types/Good';
 // eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
-  return fetch(API_URL)
-    .then(response => response.json());
+export async function getAll(): Promise<Good[]> {
+  const response = await fetch(API_URL);
+
+  if (!response.ok) {
+    return Promise.reject(
+      new Error(`${response.status} - ${response.statusText}`),
+    );
+  }
+
+  if (!response.headers.get('content-type')?.includes('application/json')) {
```

Comment:
Like in tutorials in right place, man you're cool for doing this

Reply:
I just made it the same as in Misha's video

Reply:
why do you need it here?)
Just catch your error here, it would be enough